### PR TITLE
Challenges Complete

### PR DESCRIPTION
**Pull request by Scodia619**

### CODE REVIEW

File: __tests__/server.test.js [hunk 1]
```diff
@@ -71,4 +71,54 @@ describe("GET /api/articles/:article_id", ()=>{
...
+        })
+    })
+    test("404: No article Found", ()=>{
+        const incVotes = {inc_votes: 1}
+        return request(app)
+        .patch("/api/articles/999")
+        .send(incVotes)
+        .then(({body})=>{
+            expect(body.msg).toBe("Article not found")
+        })
```
Comment (ghost) [inline __tests__/server.test.js]: One more test to go here, what about patching to a bad url? eg /api/articles/banana what would/should happen?

File: models/articles.model.js [hunk 2]
```diff
@@ -10,4 +10,16 @@ exports.getArticleById = (id) => {
         }
         return rows[0]
     })
-}
\ No newline at end of file
+}
+
+exports.updateArticleById = (id, increment) => {
+    return db.query('UPDATE articles SET votes = COALESCE(votes, 0) + $2 WHERE article_id = $1 RETURNING *', [id, increment]).then(({rows})=>{
```
Comment (ghost) [inline models/articles.model.js]: Coalesce here isn't really needed. If there are no votes on the patch request we don't need to make a database query.

**Review by ghost — CHANGES_REQUESTED** [review]
Just a test missing here, check what happens when patching to a bad ID.

**Review by nc-mick-fay — APPROVED** [review]
Happy to approve this pull request now that test has been added. Merge into main when you're ready

**Pull request merged**